"expected ‘)’ before ‘>’ token" make error with ubuntu trusty (14)

This was encountered when trying to fix the travis Ci build in #45. It was resolved by updating ubuntu to xenial (16).
I've put the relevant travis config (and thus steps for replication) below, and the relevant make errors below that, just in case [the direct link to the travis build](https://travis-ci.org/mlpack/models/jobs/646449834?utm_medium=notification&utm_source=github_status) doesn't work.
___
```sudo: required
dist: trusty
language: cpp

before_install:
  - sudo apt-get update -qq
  - sudo apt-get install -qq --no-install-recommends cmake binutils-dev libopenblas-dev liblapack-dev build-essential libboost-all-dev
  - curl -O http://masterblaster.mlpack.org:5005/armadillo-8.400.0.tar.gz -o armadillo-8.400.0.tar.gz && tar xvzf armadillo-8.400.0.tar.gz && cd armadillo-8.400.0
  - cmake . && make && ARMADILLO_INCLUDE_DIR=$(pwd)/include
  - cd $TRAVIS_BUILD_DIR && git clone https://github.com/mlpack/mlpack.git --depth 1
  - cd mlpack && mkdir mlpack_build && cd mlpack_build && cmake -DUSE_OPENMP=OFF -DARMADILLO_INCLUDE_DIR=$ARMADILLO_INCLUDE_DIR -DBUILD_PYTHON_BINDINGS=OFF -DBUILD_TESTS=OFF .. && make -j2 && sudo make install

install:
  - cd $TRAVIS_BUILD_DIR && mkdir build && cd build && cmake -DUSE_OPENMP=OFF -DARMADILLO_INCLUDE_DIR=$ARMADILLO_INCLUDE_DIR ..
script:
  - make -j2

notifications:
  email:
    - mlpack-git@lists.mlpack.org
  irc:
    channels:
      - "chat.freenode.net#mlpack"
    on_success: change
on_failure: always

```
___
```Scanning dependencies of target DigitRecognizerCNN

[  7%] Building CXX object Kaggle/DigitRecognizer/CMakeFiles/DigitRecognizer.dir/src/DigitRecognizer.cpp.o

[ 14%] Building CXX object Kaggle/DigitRecognizerCNN/CMakeFiles/DigitRecognizerCNN.dir/src/DigitRecognizerCNN.cpp.o

In file included from /usr/local/include/mlpack/methods/ann/layer/layer.hpp:41:0,

                 from /home/travis/build/mlpack/models/Kaggle/DigitRecognizerCNN/src/DigitRecognizerCNN.cpp:19:

/usr/local/include/mlpack/methods/ann/layer/weight_norm.hpp:68:76: error: expected ‘)’ before ‘>’ token

   WeightNorm(LayerTypes<CustomLayers...> layer = LayerTypes<CustomLayers...>());

                                                                            ^

/usr/local/include/mlpack/methods/ann/layer/weight_norm.hpp:68:73: error: expected ‘;’ at end of member declaration

   WeightNorm(LayerTypes<CustomLayers...> layer = LayerTypes<CustomLayers...>());

                                                                         ^

/usr/local/include/mlpack/methods/ann/layer/weight_norm.hpp:68:76: error: expected unqualified-id before ‘>’ token

   WeightNorm(LayerTypes<CustomLayers...> layer = LayerTypes<CustomLayers...>());

                                                                            ^

/usr/local/include/mlpack/methods/ann/layer/weight_norm.hpp:68:61: error: template argument 1 is invalid

   WeightNorm(LayerTypes<CustomLayers...> layer = LayerTypes<CustomLayers...>());

                                                             ^

In file included from /usr/local/include/mlpack/methods/ann/layer/weight_norm.hpp:202:0,

                 from /usr/local/include/mlpack/methods/ann/layer/layer.hpp:41,

                 from /home/travis/build/mlpack/models/Kaggle/DigitRecognizerCNN/src/DigitRecognizerCNN.cpp:19:

/usr/local/include/mlpack/methods/ann/layer/weight_norm_impl.hpp:29:1: error: prototype for ‘mlpack::ann::WeightNorm<InputDataType, OutputDataType, CustomLayers>::WeightNorm(mlpack::ann::LayerTypes<CustomLayers ...>)’ does not match any in class ‘mlpack::ann::WeightNorm<InputDataType, OutputDataType, CustomLayers>’

 WeightNorm<InputDataType, OutputDataType, CustomLayers...>::WeightNorm(

 ^

In file included from /usr/local/include/mlpack/methods/ann/layer/layer.hpp:41:0,

                 from /home/travis/build/mlpack/models/Kaggle/DigitRecognizerCNN/src/DigitRecognizerCNN.cpp:19:

/usr/local/include/mlpack/methods/ann/layer/weight_norm.hpp:68:3: error: candidate is: mlpack::ann::WeightNorm<InputDataType, OutputDataType, CustomLayers>::WeightNorm(mlpack::ann::LayerTypes<CustomLayers ...>, ...)

   WeightNorm(LayerTypes<CustomLayers...> layer = LayerTypes<CustomLayers...>());

   ^

In file included from /usr/local/include/mlpack/methods/ann/layer/layer.hpp:41:0,

                 from /home/travis/build/mlpack/models/Kaggle/DigitRecognizer/src/DigitRecognizer.cpp:19:

/usr/local/include/mlpack/methods/ann/layer/weight_norm.hpp:68:76: error: expected ‘)’ before ‘>’ token

   WeightNorm(LayerTypes<CustomLayers...> layer = LayerTypes<CustomLayers...>());

                                                                            ^

/usr/local/include/mlpack/methods/ann/layer/weight_norm.hpp:68:73: error: expected ‘;’ at end of member declaration

   WeightNorm(LayerTypes<CustomLayers...> layer = LayerTypes<CustomLayers...>());

                                                                         ^

/usr/local/include/mlpack/methods/ann/layer/weight_norm.hpp:68:76: error: expected unqualified-id before ‘>’ token

   WeightNorm(LayerTypes<CustomLayers...> layer = LayerTypes<CustomLayers...>());

                                                                            ^

/usr/local/include/mlpack/methods/ann/layer/weight_norm.hpp:68:61: error: template argument 1 is invalid

   WeightNorm(LayerTypes<CustomLayers...> layer = LayerTypes<CustomLayers...>());

                                                             ^

In file included from /usr/local/include/mlpack/methods/ann/layer/weight_norm.hpp:202:0,

                 from /usr/local/include/mlpack/methods/ann/layer/layer.hpp:41,

                 from /home/travis/build/mlpack/models/Kaggle/DigitRecognizer/src/DigitRecognizer.cpp:19:

/usr/local/include/mlpack/methods/ann/layer/weight_norm_impl.hpp:29:1: error: prototype for ‘mlpack::ann::WeightNorm<InputDataType, OutputDataType, CustomLayers>::WeightNorm(mlpack::ann::LayerTypes<CustomLayers ...>)’ does not match any in class ‘mlpack::ann::WeightNorm<InputDataType, OutputDataType, CustomLayers>’

 WeightNorm<InputDataType, OutputDataType, CustomLayers...>::WeightNorm(

 ^

In file included from /usr/local/include/mlpack/methods/ann/layer/layer.hpp:41:0,

                 from /home/travis/build/mlpack/models/Kaggle/DigitRecognizer/src/DigitRecognizer.cpp:19:

/usr/local/include/mlpack/methods/ann/layer/weight_norm.hpp:68:3: error: candidate is: mlpack::ann::WeightNorm<InputDataType, OutputDataType, CustomLayers>::WeightNorm(mlpack::ann::LayerTypes<CustomLayers ...>, ...)

   WeightNorm(LayerTypes<CustomLayers...> layer = LayerTypes<CustomLayers...>());

   ^

make[2]: *** [Kaggle/DigitRecognizer/CMakeFiles/DigitRecognizer.dir/src/DigitRecognizer.cpp.o] Error 1

make[1]: *** [Kaggle/DigitRecognizer/CMakeFiles/DigitRecognizer.dir/all] Error 2

make[1]: *** Waiting for unfinished jobs....

make[2]: *** [Kaggle/DigitRecognizerCNN/CMakeFiles/DigitRecognizerCNN.dir/src/DigitRecognizerCNN.cpp.o] Error 1

make[1]: *** [Kaggle/DigitRecognizerCNN/CMakeFiles/DigitRecognizerCNN.dir/all] Error 2

make: *** [all] Error 2```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,25 +1,13 @@
-cmake_minimum_required(VERSION 2.6)
+cmake_minimum_required(VERSION 3.1.0 FATAL_ERROR)
 project(models C CXX)
 
 # First, define all the compilation options.
 # We default to debugging mode for developers.
-option(FORCE_CXX11
-    "Don't check that the compiler supports C++11, just assume it.  Make sure to specify any necessary flag to enable C++11 as part of CXXFLAGS."
-    OFF)
 option(DOWNLOAD_ENSMALLEN "If ensmallen is not found, download it." ON)
 option(USE_OPENMP "If available, use OpenMP for parallelization." OFF)
 
-# If we are not forcing C++11 support, check that the compiler supports C++11
-# and enable it.
-if (NOT FORCE_CXX11)
-  include(CMake/CXX11.cmake)
-  check_for_cxx11_compiler(HAS_CXX11)
-
-  if(NOT HAS_CXX11)
-    message(FATAL_ERROR "No C++11 compiler available!")
-  endif()
-  enable_cxx11()
-endif ()
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 # Include modules in the CMake directory.
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/CMake")

